plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -467,37 +629,59 @@ open class DefaultPageCreator(
             }
         }
 
-    private fun DocumentableContentBuilder.functionsBlock(name: String, list: Collection<DFunction>) = divergentBlock(
+    private fun DocumentableContentBuilder.functionsBlock(
+        name: String,
+        toggleableContentType: ToggleableContentType,
+        list: Collection<DFunction>,
+        isVisibleHeader: Boolean = false
+    ) = divergentBlock(
         name,
         list.sorted(),
         ContentKind.Functions,
-        extra = mainExtra + SimpleAttr.header(name)
+        extra = mainExtra + ToggleableContentTypeExtra(toggleableContentType),
+        isVisibleHeader = isVisibleHeader
     )
 
     private fun DocumentableContentBuilder.propertiesBlock(
         name: String,
+        toggleableContentType: ToggleableContentType,
         list: Collection<DProperty>,
-        sourceSets: Set<DokkaSourceSet>
+        sourceSets: Set<DokkaSourceSet>,
+        isVisibleHeader: Boolean = false
     ) {
+        val groupedElements = list.groupBy { Pair(it.name, it.isExtension()) }.toList()
+        val sortedGroupedElements =
+            groupedElements.sortedWith(compareBy<Pair<Pair<String, Boolean>, List<DProperty>>, String>(nullsLast(String.CASE_INSENSITIVE_ORDER)) { it.first.first }.thenBy { it.first.second })
```

Comment:
Because of `<Pair<Pair<String, Boolean>, .., String>` and `it.first.second` it's quite difficult to understand what it's being sorted by

Can you please break it down or deconstruct it?